ql/experimental/variancegamma/fftengine.hpp

```diff
@@ -54,7 +54,7 @@ namespace QuantLib {
 
         void precalculate(const std::vector<boost::shared_ptr<Instrument> >& optionList);
         virtual std::auto_ptr<FFTEngine> clone() const = 0;
-
+        virtual ~FFTEngine() {}
```

Comment:
A virtual destructor for this class is already inherited from `VanillaOption::engine`, so I'd let the compiler provide it.  The same goes for the other pricing engines.